File: flask_app/model/entities/trip/Trip.py
from dataclasses import dataclass
from typing import List

from model.entities.costs.Costs import Costs
from model.entities.location.Location import Location
from model.entities.segment.Segment import Segment
from model.enums.mobi_score.MobiScore import MobiScore
from model.enums.mode.TripMode import TripMode


@dataclass
class Trip:
    start_location: Location
    end_location: Location
    trip_mode: TripMode
    segments: List[Segment]
    duration: float
    distance: float
    costs: Costs
    mobi_score: MobiScore

File: flutter_frontend/multimodal_routeplanner/lib/03_domain/entities/Trip.dart
import 'package:multimodal_routeplanner/03_domain/entities/Costs/Costs.dart';
import 'package:multimodal_routeplanner/03_domain/entities/Segment.dart';

import 'MobilityMode.dart';

class Trip {
  final double distance;
  final double duration;
  final Costs costs;
  final List<Segment> segments;
  final String mode;
  final String mobiScore;

  Trip(
      {required this.distance,
      required this.duration,
      required this.costs,
      required this.segments,
      required this.mode,
      required this.mobiScore});
}
